src/components/Header/Header.js

```diff
@@ -1,30 +1,46 @@
 import React from "react";
 import { Link } from "react-router-dom";
 import styled from "styled-components";
+import { kakaoLogin } from "../Login/kakaoLogin";
+import { useAuth } from "../../hooks/useAuth";
 
 const Header = () => {
-  const token = true;
+  const { user, clearUser, isInitialized, isAuthenticated } = useAuth();
+  const logout = () => {
+    clearUser();
+    alert(`로그아웃 되었습니다.`);
+  };
 
   return (
     <HeaderFlex>
-      <Link to="/">LOGO</Link>
-      <CategoryFlex>
-        <CategoryName>
-          <Link to="/">예약하기</Link>
-        </CategoryName>
-        <CategoryName>
-          <Link to="Matching">매칭하기</Link>
-        </CategoryName>
-      </CategoryFlex>
-      <CategoryFlex>
-        <LoginButton>{!token ? "로그인 및 회원가입" : "로그아웃"}</LoginButton>
-        {token && (
-          <ProfileButton>
-            <MenuIcon src="images/Nav/menu_FILL0_wght400_GRAD0_opsz48.png"></MenuIcon>
-            <ProfileImg />
-          </ProfileButton>
-        )}
-      </CategoryFlex>
+      {isInitialized && (
+        <>
+          <Link to="/">LOGO</Link>
+          <CategoryFlex>
+            <CategoryName>
+              <Link to="/">예약하기</Link>
+            </CategoryName>
+            <CategoryName>
+              <Link to="Matching">매칭하기</Link>
+            </CategoryName>
+          </CategoryFlex>
+          <CategoryFlex>
+            {!isAuthenticated ? (
+              <LoginButton onClick={() => kakaoLogin()}>
```

Comment:
```suggestion
              <LoginButton onClick={kakaoLogin}>
```
인자로 전달하는 값이 없다면 콜백함수 형태로 주지 않아도 됩니다.